grakn-kb/src/main/java/ai/grakn/kb/internal/concept/ElementFactory.java

```diff
@@ -154,8 +155,7 @@ public RoleImpl buildRole(VertexElement vertex, Role type){
         try {
             type = getBaseType(vertexElement);
         } catch (IllegalStateException e){
-            LOG.warn("Invalid vertex [" + vertexElement + "] due to " + e.getMessage(), e);
-            return Optional.empty();
+            throw TemporaryWriteException.indexOverlap(vertexElement.element(), e);
```

Comment:
This method now always returns something! It doesn't have to be optional if we want.